csrc/dynamic_transform.cpp

```diff
@@ -308,20 +393,36 @@ void DynamicTransformConcretizer::mutate(TensorView* tv) {
             ". IterDomain was expected.");
       }
 
-      // If none of the output IDs is symbolic, nothing to concretize
+      // If all inputs are concrete, all outputs should be concrete, and there
+      // is nothing to concretize.
       if (std::all_of(
-              expr->outputs().begin(), expr->outputs().end(), [](Val* output) {
+              expr->inputs().begin(), expr->inputs().end(), [](Val* output) {
```

Comment:
Here, https://github.com/NVIDIA/Fuser/pull/258#discussion_r1192899457, you mentioned:

> We may have a concrete input to resize that leads to symbolic output, whenever we have non constant pad widths.

What you mentioned seems to make sense to me, but this assertion seems to conflict with it, no?

Reply:
Yes you're right. The quote would apply to `?S1 = resize(iS0, s1, s2)` since we can't know the output extent when the pad widths are symbolic. I think we can remove this check; the effect would be that every expression between root and rfactor will be processed. But I don't think that's really an issue since if all inputs and outputs are concrete, the rest of this loop has no effect.